golem_messages/utils.py

```diff
@@ -2,17 +2,18 @@
 import copy
 from uuid import UUID
 
-from eth_utils import to_checksum_address, keccak
+from eth_utils import to_checksum_address, keccak, hexstr_if_str
 
 from golem_messages import message
 
 
 def pubkey_to_address(pubkey: str) -> str:
     """
-    convert a hex-encoded pubkey into a checksummed address
+    convert pubkey into a checksummed address
+    pubkey might be hex str or bytes or int
```

Comment:
No, we don't do that here. `pubkey` is a string, is typed as such and leave it this way.

Reply:
why is that? can you provide some rationale?

Reply:
throw-anything-as-an-argument-and-hope-for-the-best just doesn't seem like a maintainable approach. Easier to write and read code when you know the type of the variable.
Also better integration with tools like static analyzer.

Reply:
This is how it works in eth lib used by us here. Its a standard there. We support three types of input, not 'anything': string, hexstr and int.

It is usable in such a form but I can introduce dedicated params...

Reply:
But we don't need to support three types. At this point almost everywhere we're working on strings. So just provide strings and that's it.

Reply:
I needed it to support bytes representation and it seems to me I've found a pretty concise and effective way to achieve it. Much better, I think, that you can observe 5 lines below -- just expand this view and you'll see, that next util function already supports different types on the single input parameter.

This is "The Python Way":
* numpy/scipy [prominent example](https://docs.scipy.org/doc/numpy/user/basics.creation.html#converting-python-array-like-objects-to-numpy-arrays)
* pandas [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html)
* Web3 eth API [here](https://web3py.readthedocs.io/en/stable/web3.eth.html#web3.eth.Eth.sendTransaction)

One thing you've noticed is to be corrected -- the parameter type.

Reply:
Python way is also writing unmaintainable and illegible code, doesn't mean we should follow that.
This is the first `Union` type hint in this repository - meaning, what I said earlier, we don't do that here. We should be consistent within the same project and this case is way too minor to break current conventions.

Reply:
I don't approve this change but I have nothing else to add and am dismissing my review since this won't be resolved.